src/dataflow/src/render/reduce.rs

```diff
@@ -1253,6 +1323,39 @@ where
                 },
                 x => panic!("Invalid argument to AggregateFunc::Dummy: {:?}", x),
             },
+            AggregateFunc::SumAPD => match datum {
+                Datum::APD(n) => {
+                    let (accum, pos_infs, neg_infs, nans) = if n.0.is_infinite() {
```

Comment:
nit: I sometimes like to do something like:

```
diff --git a/src/dataflow/src/render/reduce.rs b/src/dataflow/src/render/reduce.rs
index 108c42f0b..98bfb48e0 100644
--- a/src/dataflow/src/render/reduce.rs
+++ b/src/dataflow/src/render/reduce.rs
@@ -1325,18 +1325,11 @@ where
             },
             AggregateFunc::SumAPD => match datum {
                 Datum::APD(n) => {
-                    let (accum, pos_infs, neg_infs, nans) = if n.0.is_infinite() {
-                        if n.0.is_negative() {
-                            (DecNum::<26>::zero(), 0, 1, 0)
-                        } else {
-                            (DecNum::<26>::zero(), 1, 0, 0)
-                        }
-                    } else if n.0.is_nan() {
-                        (DecNum::<26>::zero(), 0, 0, 1)
-                    } else {
-                        // Convert out 13-width datum to a 26-width value for
-                        // accumulation.
-                        (n.0.to_width::<26>().unwrap(), 0, 0, 0)
+                    let (accum, pos_infs, neg_infs, nans) = match (n.0.is_infinite(), n.0.is_negative(), n.0.is_nan()) {
+                        (true, false, _) => (DecNum::<26>::zero(), 1, 0, 0),
+                        (true, true, _) => (DecNum::<26>::zero(), 0, 1, 0),
+                        (_, _, true) => (DecNum::<26>::zero(), 0, 0, 1),
+                        _ => (n.0.to_width::<26>::zero(), 0, 0, 0),
                     };

                     AccumInner::APD {
```

for this type of logic. Caveat: not tested. I typed it up but im getting complaints about cargo not knowing about rust-dec

Reply:
Oh that is really nice! ty for the pointer